design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,361 @@
+
+    estimateShippingInformationForNegotiableQuote(
+        addressInformation: NegotiableQuoteShippingInformationInput!
+    ): EstimateShippingInformationForNegotiableQuoteOutput @doc(description:"Get list of payment options and the order totals by address and shipping info")
+
+    setShippingMethodToNegotiableQuote(
+        negotiableQuoteShippingMethod: NegotiableQuoteShippingMethodInput!
+    ): setShippingMethodToNegotiableQuoteOutput @doc(description:"Set negotiable quote shipping method")
+
+    setCouponCodeToNegotiableQuote(
+        negotiableQuoteCouponCode: NegotiableQuoteCouponCodeInput!
+    ): SetCouponCodeToNegotiableQuoteOutput @doc(description:"Set negotiable quote coupon code")
+
+    removeCouponCodeFromNegotiableQuote(
+        cartId: String!
+    ): RemoveCouponCodeFromNegotiableQuoteOutput @doc(description:"Remove negotiable quote coupon code")
+
+    setPaymentInformationForNegotiableQuote(
+        paymentInformation: NegotiableQuotePaymentInformationInput!
+    ): SetPaymentInformationForNegotiableQuoteOutput! @doc(description:"Sets payment information and the billing address for the negotiable quote")
+
+    createOrderWithPaymentInfoNegotiableQuote(
+        paymentInformation: NegotiableQuotePaymentInformationInput!
+    ): CreateOrderWithPaymentInfoNegotiableQuoteOutput! @doc(description:"Assigns payment information and the billing address for the negotiable quote to create order")
+
+    createOrderWithGiftCardForNegotiableQuote(
+        giftCardAccountData: NegotiableQuoteGiftCardAccountDataInput!
+    ): CreateOrderWithGiftCardForNegotiableQuoteOutput! @doc(description:"Assigns GiftCard information and cart ID for the negotiable quote to create order")
+
+    removeGiftCardFromNegotiableQuote(
+        giftCardsData: NegotiableQuotegiftCardsInput!
+    ): RemoveGiftCardFromNegotiableQuoteOutput! @doc(description:"Removes a gift card that has been applied to a negotiable quote")
+}
+
+type Customer {
+        negotiable_quotes(
+        page_size: Int = 20,
+        current_page: Int = 1,
+        filter: NegotiableQuoteFilterInput
+    ): [NegotiableQuote] @doc(description: "Get negotiable quotes of a customer")
+}
+
+type NegotiableQuote {
+    billing_address: BillingAddress! @doc(description: "Billing address of negotiable quote")
+    attachements: [AttachmentContent] @doc(description: "Negotiable quote Attachements")
+    comments: [NegotiableQuoteComment] @doc(description: "Negotiable quote comments")
+    totals: TotalInfo @doc(description: "Negotiable quote totals info.")
+    payment_info: NegotiableQuotePaymentInfo! @doc(description: "Negotiable quote payment info.")
+}
+
+input NegotiableQuoteFilterInput {
+    ids: FilterEqualTypeInput @doc(description: "Filter Customer Negotibale quotes with an negotiable quote ID or list of negotiable quote IDs")
+    name: FilterMatchTypeInput @doc(description: "Filter by display name of the negotiable quote")
+}
+
+type BillingAddress {
```

Comment:
Is there a reason not to reuse `CartAddressInterface`?

Reply:
The BillingAddress has `email` and `save_in_addressbook` fields which the CartAddressInterface lacks. can implement CartAddressInterface in our schema? To add the missing fields?

Reply:
1. Why `email` should be a property of `CartAddress`?
2. `save_in_addressbook` is not included there on purpose because it does not make sense in APIs. It can be rendered in the UI, but we should have dedicated mutation to save it, not as a side effect

Reply:
The Billing address field has been removed according to https://github.com/magento/architecture/pull/381#discussion_r434664761